[Copter] [SITL] [Linux] build error due to std::uint16_t

## Bug report

**Issue details**

In the CanSocketIface.cpp files (https://github.com/search?q=repo%3AArduPilot%2Fardupilot%20std%3A%3Auint16_t&type=code), there is usage of both `std::uint16_t` and just `uint16_t`. The use of `std::uint16_t` causes a compilation error, since under later gcc versions `std::uint16_t` requires the `<cstdint>` include, which may not be portable to other compiler versions.

I believe the occurrence of `std::uint16_t` should be removed and just `uint16_t` should be used like in the rest of the Ardupilot codebase.

**Version**
master branch latest commit: [15601e4](https://github.com/ArduPilot/ardupilot/commit/15601e413982d6f362a870d67586e1510c5ad2b1)

**Platform**
[  ] All
[  ] AntennaTracker
[x] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
_What type of airframe (flying wing, glider, hex, Y6, octa etc)_
N/A

**Hardware type**
_What autopilot hardware was used? (Pixhawk, Cube, Pixracer, Navio2, etc)_
SITL

**Logs**
```
[621/974] Compiling libraries/AP_HAL_SITL/CANSocketIface.cpp
../../libraries/AP_HAL_SITL/CANSocketIface.cpp:262:42: error: ‘uint16_t’ in namespace ‘std’ does not name a type; did you mean ‘wint_t’?
  262 |                               const std::uint16_t num_configs)
      |                                          ^~~~~~~~
      |                                          wint_t
compilation terminated due to -Wfatal-errors.
```

--- a/libraries/AP_HAL_Linux/CANSocketIface.cpp
+++ b/libraries/AP_HAL_Linux/CANSocketIface.cpp
@@ -221,7 +221,7 @@ void CANIface::_poll(bool read, bool write)
 }
 
 bool CANIface::configureFilters(const CanFilterConfig* const filter_configs,
-                              const std::uint16_t num_configs)
+                              const uint16_t num_configs)
 {
     if (filter_configs == nullptr || mode_ != FilteredMode) {
         return false;

--- a/libraries/AP_HAL_SITL/CANSocketIface.cpp
+++ b/libraries/AP_HAL_SITL/CANSocketIface.cpp
@@ -261,7 +261,7 @@ void CANIface::_poll(bool read, bool write)
 }
 
 bool CANIface::configureFilters(const CanFilterConfig* const filter_configs,
-                              const std::uint16_t num_configs)
+                              const uint16_t num_configs)
 {
 #if 0
     if (filter_configs == nullptr || mode_ != FilteredMode) {

